kie-wb-tests/kie-wb-tests-gui/src/test/java/org/kie/wb/selenium/ui/StandalonePerspectivesIntegrationTest.java

```diff
@@ -0,0 +1,121 @@
+    private final int PERSPECTIVE_LOADING_TIMEOUT = 30;
+
+    private static final By HEADER = By.id("workbenchHeaderPanel");
+    private static final Map<String, By> perspectiveToElement = new HashMap<String, By>() {
+        {
+            //put("FormDisplayPerspective", ); // How to test this? What does this perspective do?
+            //put("StandaloneEditorPerspective", ); // How to test this? I need an additional parameter with a path to some asset.
+
+            put("AdministrationPerspective", By.className("qe-list-bar-header-Repository-Editor"));
+            put("AdminPagePerspective", By.id("admin-page-title"));
+            put("AppsPerspective", By.cssSelector("[title='Create Directory']"));
+            put("Asset Management", By.className("qe-list-bar-header-Repository-Configuration"));
+            put("AuthoringPerspective", By.className("fa-refresh"));
+            put("AuthoringPerspectiveNoContext", By.cssSelector("[title='Project Explorer']"));
+            put("ContentManagerPerspective", By.className("qe-list-bar-header-Content-Manager"));
+            put("DashboardPerspective", ByJQuery.selector("span:contains('Processes & Tasks Dashboard')"));
+            put("DataSetAuthoringPerspective", By.className("qe-list-bar-header-Data-Set-Authoring-Home"));
+            put("DataSourceManagementPerspective", By.className("qe-list-bar-header-DataSource-Explorer"));
+            put("ExecutionErrors", By.className("qe-list-bar-header-Execution-Errors"));
+            put("GuvnorM2RepoPerspective", By.className("qe-list-bar-header-M2-Repository-Content"));
+            put("HomePerspective", ByJQuery.selector("h2:contains('The Knowledge Life Cycle')"));
+            put("LibraryPerspective", By.id("welcome"));
+            put("PlugInAuthoringPerspective", ByJQuery.selector("h3:contains('Plugin Explorer')"));
+            put("PreferencesCentralPerspective", By.className("preferences-editor"));
+            put("ProcessDefinitions", By.className("qe-list-bar-header-Process-Definitions"));
+            put("ProcessInstances", By.className("qe-list-bar-header-Process-Instances"));
+            put("Requests", By.className("qe-list-bar-header-Requests-List"));
+            put("SecurityManagementPerspective", By.className("qe-list-bar-header-Security-management"));
+            put("ServerManagementPerspective", By.className("fa-refresh"));
+            put("SocialHomePagePerspective", By.className("qe-list-bar-header-Recent-Assets"));
+            put("TaskAdmin", By.className("qe-list-bar-header-Task-Administration-List"));
+            put("Tasks", By.className("qe-list-bar-header-Tasks-List"));
+            put("UserHomePagePerspective", By.className("fa-home"));
+            put("WiresGridsDemoPerspective", By.className("qe-list-bar-header-Grids"));
+            put("WiresTreesPerspective", By.className("qe-list-bar-header-Palette"));
+        }
+    };
+
+    @Test
+    public void loadStandalonePerspectives() {
```

Comment:
Unfortunately in this case we can't use `@Parametrized` as it requires different JUnit Runner. Since we're using `@Drone` to setup WebDriver, we're forced to use `@RunWith(Arquillian.class)` in the [parent test class](https://github.com/kiegroup/kie-wb-distributions/blob/master/kie-wb-tests/kie-wb-tests-gui/src/main/java/org/kie/wb/selenium/model/KieSeleniumTest.java#L28).

There's another issue why we went for 1 test with for loop: the need for `@BeforeClass` setup (logging in once before starting to load all perspectives), which is impossible with the current version of Drone, which is not initialized at the point when `@BeforeClass` is running. I heard this has been fixed in Drone 2 but didn't want to do too many changes at once. We'll do Drone/Graphen/Selenium version update (and related Firefox update in CI images) next. Larger refactoring will be needed anyway once the Megamenu proposal is implemented.

Take this initial implementation as a medium for communication with some details to be worked out:
* Which perspectives do we want to test in this way?
* Do we want to remove some persps from the workbench?
* What do we do with perspective loading times sometimes exceeding 30 seconds?
* Do we want to test more stuff than that the perspective was loaded?
* `header=AppNavBar` only shows useful stuff in AuthoringPerspective, AuthoringPerspectiveNoContext and AdministrationPerspective, so we probably don't need to check the effect of this parameter in other perspectives, right?

 After @manstis provides some clarification, we'll refactor this to "1 test per perspective" as you suggested, which although it requires more boilerplate, will give the benefit of more granular coverage and not have the disadvantage of "fail cause the following things not to be tested".